brainrender/actor.py

```diff
@@ -41,47 +55,62 @@ def __str__(self):
 
         return buf.getvalue()
 
-    def __rich_console__(self, *args):
-        rep = pi.Report(
-            title=f"[b]brainrender.Actor: ", color=salmon, accent=orange,
-        )
-
-        rep.add(f"[b {orange}]name:[/b {orange}][{mocassin}] {self.name}")
-        rep.add(f"[b {orange}]type:[/b {orange}][{mocassin}] {self.br_class}")
-        rep.line()
-        rep.add(
-            f"[{orange}]center of mass:[/{orange}][{mocassin}] {self.mesh.centerOfMass().astype(np.int32)}"
-        )
-        rep.add(
-            f"[{orange}]number of vertices:[/{orange}][{mocassin}] {len(self.mesh.points())}"
-        )
-        rep.add(
-            f"[{orange}]dimensions:[/{orange}][{mocassin}] {np.array(self.mesh.bounds()).astype(np.int32)}"
-        )
-        rep.add(f"[{orange}]color:[/{orange}][{mocassin}] {self.mesh.color()}")
-
-        yield rep
-
     @classmethod
     def make_actor(cls, mesh, name, br_class):
+        """
+            Make an actor from a given mesh
+        """
         return cls(mesh, name=name, br_class=br_class)
 
     def make_label(self, atlas):
+        """
+            Create a new Actor with a sphere and a text
+            labelling this actor
+        """
         labels = make_actor_label(
```

Comment:
Also, an actor will always be referenced to a scene and its atlas, so probably atlas should just be passed to the actor init

Reply:
>This usage of make_actor_label is a bit funny in that it's functional coding on something that it's clearly object-oriented - why not having it as a method of the class?

It's a relative long (80LOC) and ugly bit of code, so I wanted to hide it away from the class definition. 
I've moved the function to `actor` (from `_actor`) to remove the need for `_actor`, but left it out of the class for now.
I know it's now very elegant but might be okay for now?

> Also, an actor will always be referenced to a scene and its atlas, so probably atlas should just be passed to the actor init

Yeah, that could be done. Though the atlas attribute is only used to create the labels, and even then only to make sure that the labels are in the correct hemisphere... 

This whole label's business is quite ugly for both handling the creation and the rendering of the labels...

Reply:
I do agree that this bit of code, having the atlas just for the labels creation, looks a bit cumbersome...I will see if I can come out with more complete suggestions on how this could be improved

Reply:
okay thanks!

Reply:
I'd be tempted to scrap the whole thing to be honest